relay-sampling/src/lib.rs

```diff
@@ -1165,6 +1165,43 @@ mod sample_rate_as_string {
     }
 }
 
+mod sampled_as_string {
+    use super::*;
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<bool>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = match Option::<Cow<'_, str>>::deserialize(deserializer)? {
+            Some(value) => value,
+            None => return Ok(None),
+        };
+
+        if value == "true" {
+            Ok(Some(true))
+        } else if value == "false" {
+            Ok(Some(false))
+        } else {
+            Err(serde::de::Error::custom(
+                "the `sampled` value can only be `true` or `false`",
+            ))
+        }
+    }
+
+    pub fn serialize<S>(value: &Option<bool>, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
- Let's support deserializing from both `bool` and `string`, then always serialize with the default serializer.
- If we want to be extra lenient, we could ignore all errors and make them `None`.

Reply:
There's a way to deserialize this without going through `Value` by implementing a [visitor](https://serde.rs/impl-deserialize.html#the-visitor-trait). I can help with that tomorrow. 

As for serialization, I'd still ask to serialize directly as boolean. You can remove `fn serialize` entirely and just define `deserialize_with`.